src/fsharp/MethodOverrides.fs

```diff
@@ -420,7 +436,110 @@ module DispatchSlotChecking =
                     | meth :: _ when meth.IsFinal -> errorR(Error(FSComp.SR.tcCannotOverrideSealedMethod((sprintf "%s::%s" (meth.ApparentEnclosingType.ToString()) (meth.LogicalName))), m))
                     | _ -> ()
 
+    /// Get default slot flags for the given method.
+    let inline GetDefaultDispatchSlotFlags (minfo: MethInfo) =
+        if minfo.IsAbstract then RequiredSlotFlags.None else RequiredSlotFlags.Optional
+
+    /// Get a collection of interface slots that are considered possible overrider (aka. override by) slots for the given method from the given type and implied types.
+    let GetPossibleInterfaceOverriderDispatchSlots (infoReader: InfoReader) m minfo reqdTy impliedTys =
+        let g = infoReader.g
+        let amap = infoReader.amap
+
+        if isInterfaceTy g reqdTy then
+            impliedTys
+            |> List.choose (fun ty ->
+                if isInterfaceTy g ty && not (typeEquiv g reqdTy ty) && TypeFeasiblySubsumesType 0 g amap m reqdTy CanCoerce ty then
+                    // We are only looking for IL overriders because F# does not support creating your own default interface methods.
+                    match TryFindILIntrinisicOverriderMethInfo infoReader m AccessibleFromSomewhere ty minfo with
+                    | Some minfo2 -> Some (ty, RequiredSlot (minfo, GetDefaultDispatchSlotFlags minfo2))
+                    | _ -> None
+                else
+                    None)
+        else
+            []
+
+    /// Get a collection of slots for the given interface type.
+    let GetInterfaceDispatchSlots (infoReader: InfoReader) m availImpliedInterfaces reqdTyInfos reqdTy =
+        let g = infoReader.g
+        let amap = infoReader.amap
+
+        let dimFeatureError = TryGetRuntimeOrLanguageSupportsFeatureError infoReader m LanguageFeature.DefaultInterfaceMethodsInterop
+
+        [ if isInterfaceTy g reqdTy then
+            // Check if the interface has an inherited implementation
+            // If so, you do not have to implement all the methods - each
+            // specific method is "optionally" implemented.
+            let isOptional = ListSet.contains (typeEquiv g) reqdTy availImpliedInterfaces
+            for minfo in GetImmediateIntrinsicMethInfosOfType (None, AccessibleFromSomewhere) g amap m reqdTy do
+              if minfo.IsDefaultInterfaceMethod then
+                  dimFeatureError |> Option.iter error
+                  
+              if not minfo.IsFinal then
+                  if isOptional || not minfo.IsILMethod then
+                      yield RequiredSlot(minfo, if isOptional then RequiredSlotFlags.Optional else RequiredSlotFlags.None)
+                  else
+                      let sortedPossibleOverriderSlots =
```

Comment:
This part is a little tricky. We really want to find the top most interface hierarchies from the given `reqdTyInfos` where `minfo` 's enclosing type is part of each of those hierarchies.